### PR TITLE
added production deployment

### PR DESCRIPTION
**Pull request by rkj2096**

### CODE REVIEW

**Review by nileshbhosale — APPROVED** [review]

File: .github/workflows/devdeploy.yml [hunk 1]
```diff
@@ -27,11 +27,25 @@ jobs:
       run: |
         python -m pip install --upgrade pip
         pip install -r requirements.txt
+    - name: Subtitute enviornment vars in app.yaml
+      run: |
+        echo "$(envsubst < app.yaml)" > app.yaml
+      env:
+        DJANGO_ENV: stage
```
Comment (vyasakanksha) [inline .github/workflows/devdeploy.yml:34]: this should be `DJANGO_ENV: stage` right?

File: .github/workflows/devdeploy.yml [hunk 2]
```diff
@@ -27,11 +27,25 @@ jobs:
...
-      run: ./.github/scripts/decrypt_secrets.sh
+      run: |
+        mkdir secrets
+        gpg --quiet --batch --yes --decrypt --passphrase="$CLOUD_STORAGE_SECRET" \
+        --output secrets/tapestry-pooling-cloud-storage-credentials.json \
+        tapestry-pooling-cloud-storage-credentials-stage.json.gpg
       env:
-        CLOUD_STORAGE_VIEW_SECRET: ${{ secrets.CLOUD_STORAGE_VIEW_SECRET }}
-        CLOUD_STORAGE_CREATE_SECRET: ${{ secrets.CLOUD_STORAGE_CREATE_SECRET }}
+        CLOUD_STORAGE_SECRET: ${{ secrets.CLOUD_STORAGE_SECRET_STAGE }}
```
Comment (vyasakanksha) [inline .github/workflows/devdeploy.yml:48]: and here?

File: rest/forms/labconfiguration.py [hunk 3]
```diff
@@ -1,11 +1,16 @@
...
 
 from rest.models import LabConfiguration
 from rest.util.gc_util import get_object_list
 
 
 class LabConfigurationForm(forms.ModelForm):
-    poolingmatrix_filename = forms.ChoiceField(choices=get_object_list('kirkman_matrices'))
+    poolingmatrix_filename = forms.ChoiceField(
+        choices=get_object_list(
+            'kirkman_matrices_prod' if os.environ.get('DJANGO_ENV') == 'prod' else 'kirkman_matrices'
```
Comment (vyasakanksha) [inline rest/forms/labconfiguration.py:11]: make the else also a env var

File: rest/util/gc_util.py [hunk 4]
```diff
@@ -15,12 +15,13 @@
...
+    'tapestry-pooling-cloud-storage-credentials.json'
+)
+
 def get_report_download_url(object_name):
-    service_account_file = os.path.join(
-        'secrets',
-        'tapestry-pooling-storage-object-viewer-credentials.json'
-    )
-    bucket_name = 'admin_results_report'
+    bucket_name = 'admin_results_report_prod' if os.environ.get('DJANGO_ENV') == 'prod' else 'admin_results_report'
```
Comment (vyasakanksha) [inline rest/util/gc_util.py:24]: 'admin_results_report' as env var

File: rest/util/gc_util.py [hunk 5]
```diff
@@ -45,15 +46,16 @@ def get_object_list(bucket_name):
 
 
 def get_pooling_matrix_download_url(payload=None, object_name=None):
-    url = "https://us-central1-tapestry-pooling-284109.cloudfunctions.net/tapestry-matrix-generation"
+    if os.environ.get('DJANGO_ENV') == 'prod':
+        url = 'https://us-central1-tapestry-pooling-f9f8e.cloudfunctions.net/tapestry-matrix-generation'
```
Comment (vyasakanksha) [inline rest/util/gc_util.py:50]: this url should be an env var

**Comment by vyasakanksha** [comment]
Lets also add a pull-request github action for master (like we have for dev). Overall this looks really good.

**Pull request merged**